### ISSUE TITLE
Problema na digitação do telefone com 11 dígitos

### ISSUE THREAD

**Issue by GricelRio** [claudiosanches/woocommerce-extra-checkout-fields-for-brazil#94]
O campo telefone não funciona corretamente com o 11o  dígito, acrescenta uma ? na penúltima posição: 
9999999-999?9

Wordpress: WordPress 4.8.6
Woocommerce: Versão 3.2.5
WooCommerce Extra Checkout Fields for Brazil: Versão 3.6.0
Woocommerce Pagseguro v. 2.12.5
WooCommerce Correios v. 3.7.0
Depois que instalaei o plugin Woo Checkout Field Editor Pro v. 1.26, o problema do telefone se resolveu, após desativar o woocommerce-extra-checkout-fields-for-brazil, mas aí o Checkout transparente do PagSeguro não funciona mais, só o Redirecionamento. 
Meu cliente quer Checkout transparente.

Abraço, Gricel

**Comment by claudiosanches** [comment]
Não consigo simular o problema usando os plugins que você esta falando, ainda tem algo errado com o seu site, provavelmente seu tema, melhor testar com outro.

**Comment by GricelRio** [comment]
O que me intriga é que só dá problema se ativar o WooCommerce Extra Checkout Fields for Brazil.
Esqueci de dizer o tema, desculpe, é o IDStore v 4.3.

**Comment by claudiosanches** [comment]
Sim, claro que da o problema ao ativar, porque é exatamente esse plugin introduzindo as máscaras, caso o contrário não iria ter isso.
Além que pode desativar essas mascaras no plugin e não precisar usar outro para campos extra.

**Comment by claudiosanches** [comment]
E já tentou com outro tema? Com o Storefront ou com algum tema padrão do WordPress?

**Comment by GricelRio** [comment]
Tentei desativar as máscaras, mas não teve efeito. 
Por isto acabei colocando o plugin Woo Checkout Field Editor Pro v. 1.26, aí tudo funcionou, menos o Checkout transparente.
Vou tentar outro tema, 
Dou notícias.

**Comment by claudiosanches** [comment]
Como assim não teve efeito?
Isso é introduzido apenas se esta ativando as mascaras.
Se não o problema é mascara de outro plugin ou do seu tema.

**Comment by GricelRio** [comment]
Eu desmarquei, mas continuou aparecendo a ?.

![configextrafields](https://user-images.githubusercontent.com/13592553/40456207-7b0e4aa4-5ec6-11e8-8e8e-99fcd695eff2.png)

![telacobranca](https://user-images.githubusercontent.com/13592553/40456291-ef77c47e-5ec6-11e8-93b0-276d44ecc634.png)

O doido é que a ? só aparece se eu ativar o woocommerce-extra-checkout-fields-for-brazil, mesmo desmarcando a máscara.

Pode ser alguma incompatibilidade entre o tema e o plugin?
Estou esperando a resposta do cliente para mudar o tema.  
Depois eu relato aqui se deu certo.

**Comment by claudiosanches** [comment]
@GricelRio acabei de ficar sabendo que tem conflito com o plugin do GerenciaNet, esta utilizando ele?

**Comment by GricelRio** [comment]
Olá,
Estes são os plugins:
![image](https://user-images.githubusercontent.com/13592553/40624168-4b5806ba-6281-11e8-962b-96597926da95.png)
![image](https://user-images.githubusercontent.com/13592553/40624204-86a8cf4c-6281-11e8-8a51-c5feb0aa0357.png)
![image](https://user-images.githubusercontent.com/13592553/40624214-992bb71a-6281-11e8-8386-bc75dbe1664c.png)
![image](https://user-images.githubusercontent.com/13592553/40624230-b601e332-6281-11e8-998f-980a391b6672.png)
Na verdade parei um pouco de pesquisar isto, pois estou preocupada com a questão do Pagseguro com TLS 1.2. Abri algumas issues e ninguém me respondeu.
Mas se descobrir algo, eu publico aqui. 
Muito obrigada pela ajuda.

**Comment by claudiosanches** [comment]
Tenta ir desativando plugin por plugin até deixar apenas os plugins que você tinha falado:

```
Woocommerce: Versão 3.2.5
WooCommerce Extra Checkout Fields for Brazil: Versão 3.6.0
Woocommerce Pagseguro v. 2.12.5
WooCommerce Correios v. 3.7.0
```

E ai me diga se encontrou qual causa esse problema.

**Comment by relima** [comment]
Confirmo conseguir reproduzir o erro. Parece ser mesmo um problema do plugin. Quando voce digita um numero do tipo (11) 1234-56789, na aba de pagamento por cartao do pagseguro, o plugin nao valida e apresenta mensagem de erro. Isso foi observado quando testando o modo sandbox - nao cheguei a verificar se acontece no real.

**Comment by relima** [comment]
Após pesquisar mais a fundo, o erro não aparece sempre e pode tambem estar associado ao api do pagseguro. A mensagem invalid hash aparece no log quando o erro acontece.

**Comment by espimentel** [comment]
Alguém sabe me dizer se foi resolvido e, se sim, como? Eu tbm tenho esse problema no meu site e, como não programo, não sei exatamente onde é o problema.
Obrigado!

**Comment by espimentel** [comment]
@claudiosanches fiz um novo teste aqui, pois essa semana teve 3 clientes que reclamaram dessa interrogação. Também tentei desabilitar a máscara pra ver se resolve e não resolveu. Tentei com o tema Napoli (ativo atualmente), mas, estou providenciando a troca pro tema Shopper.

Fiz um teste desativando os plugins e atualizando a página e reparei que o problema acontece quando está habilitado o plugin Webmania BR (que é o meu emissor de nota fiscal) + Extracheckout e os outros desenvolvidos por você. Entrarei em contato com o suporte deles, pra tentar uma solução o mais breve.

Obrigado!

**Comment by rickslayer** [comment]
> @claudiosanches fiz um novo teste aqui, pois essa semana teve 3 clientes que reclamaram dessa interrogação. Também tentei desabilitar a máscara pra ver se resolve e não resolveu. Tentei com o tema Napoli (ativo atualmente), mas, estou providenciando a troca pro tema Shopper.
> 
> Fiz um teste desativando os plugins e atualizando a página e reparei que o problema acontece quando está habilitado o plugin Webmania BR (que é o meu emissor de nota fiscal) + Extracheckout e os outros desenvolvidos por você. Entrarei em contato com o suporte deles, pra tentar uma solução o mais breve.
> 
> **UPDATE: Observei que tanto o plugin Extracheckout quanto o plugin webmania oferecem uma opção para habilitar desabilitar máscaras. Desabilitei no Webmania e deixei habilitado apenas do Extra CheckOut e sucesso! A interrogação não aparece. Só não funcionou a máscara da data de nascimento, mas, pra mim, ela é irrelevante nesse momento. Mesmo pra fazer uma identificação melhor do meu publico alvo, depois eu consigo tratar essa informação manualmente.**
> 
> Obrigado!

@espimentel Muito obrigado. Quando está ativo as opções do Webmania as opções do Extracheckout não surtem mais efeito nenhum. Nossa perdi um tempo lascado com esse "?". Salvou
